Mathlib/Algebra/Order/Monoid/Defs.lean

```diff
@@ -17,6 +17,80 @@ open Function
 
 variable {α : Type*}
 
+-- TODO: assume weaker typeclasses
+
+/-- An ordered (additive) monoid is a monoid with a partial order such that addition is monotone. -/
+class IsOrderedAddMonoid (α : Type*) [AddCommMonoid α] [PartialOrder α] where
+  protected add_le_add_left : ∀ a b : α, a ≤ b → ∀ c, c + a ≤ c + b
+  protected add_le_add_right : ∀ a b : α, a ≤ b → ∀ c, a + c ≤ b + c := fun a b h c ↦ by
+    rw [add_comm _ c, add_comm _ c]; exact add_le_add_left a b h c
+
+/-- An ordered monoid is a monoid with a partial order such that multiplication is monotone. -/
+@[to_additive]
+class IsOrderedMonoid (α : Type*) [CommMonoid α] [PartialOrder α] where
+  protected mul_le_mul_left : ∀ a b : α, a ≤ b → ∀ c, c * a ≤ c * b
+  protected mul_le_mul_right : ∀ a b : α, a ≤ b → ∀ c, a * c ≤ b * c := fun a b h c ↦ by
+    rw [mul_comm _ c, mul_comm _ c]; exact mul_le_mul_left a b h c
+
+section IsOrderedMonoid
+variable [CommMonoid α] [PartialOrder α] [IsOrderedMonoid α]
+
+@[to_additive]
```

Comment:
This instance and the next one are instances that always apply, so in theory they shouldn't get full priority. Since they will be used most of the time, would 900 be a good idea?